community/cloud-foundation/templates/bigquery/bigquery_dataset.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This template creates a BigQuery Dataset."""
+
+
+def generate_config(context):
+    """ Entry point for the deployment resources. """
+
+    # You can modify the roles you wish to whitelist here
+    whitelisted_roles = ['READER', 'WRITER', 'OWNER']
+
+    name = context.properties['name']
+
+    properties = {
+        'datasetReference':
+            {
+                'datasetId': name,
+                'projectId': context.env['project']
+            },
+        'location': context.properties.get('location',
+                                           'US')
+    }
+
+    optional_properties = [
+        'description',
+        'defaultTableExpirationMs'
+    ]
+
+    for prop in optional_properties:
+        if context.properties.get(prop):
+            properties[prop] = context.properties[prop]
+
+    if context.properties.get('access'):
+        # Validate the access roles
+        for access_role in context.properties.get('access'):
```

Comment:
Is there any specific reason to use  context.properties.get('access') and not context.properties['access'] ?

Reply:
If `access` is not in the schema, it will not be set.

Reply:
I understand, how the if statement works.

I suggest to use:
if 'access' in context.properties
and context.properties['access']

The main reason is to avoid using context.properties['access'] and context.properties.get('access') syntax for the same action.

Reply:
Done